modules/solid_mechanics/src/vectorpostprocessors/CrackFrontNonlocalMaterialBase.C

```diff
@@ -0,0 +1,148 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CrackFrontNonlocalMaterialBase.h"
+#include "CrackFrontDefinition.h"
+
+InputParameters
+CrackFrontNonlocalMaterialBase::validParams()
+{
+  InputParameters params = ElementVectorPostprocessor::validParams();
+  params.addRequiredParam<UserObjectName>("crack_front_definition",
+                                          "The CrackFrontDefinition user object name");
+  params.addRequiredParam<Real>("box_length", "Distance in front of crack front.");
+  params.addRequiredParam<Real>("box_height", "Distance normal to front of crack front.");
+  params.addParam<Real>("box_width", 1.0, "Distance tangent to front of crack front.");
+  params.addParam<std::string>("base_name",
+                               "Optional parameter that allows the user to define "
+                               "multiple mechanics material systems on the same "
+                               "block, i.e. for multiple phases");
+  params.set<bool>("use_displaced_mesh") = false;
+  params.set<ExecFlagEnum>("execute_on") = {EXEC_TIMESTEP_BEGIN};
+  params.addClassDescription("Computes the average material property at a crack front.");
+  return params;
+}
+
+CrackFrontNonlocalMaterialBase::CrackFrontNonlocalMaterialBase(const InputParameters & parameters,
+                                                               const std::string & property_name)
+  : ElementVectorPostprocessor(parameters),
+    _property_name(property_name),
+    _box_length(getParam<Real>("box_length")),
+    _box_width(getParam<Real>("box_width")),
+    _box_height(getParam<Real>("box_height")),
+    _base_name(isParamValid("base_name") ? getParam<std::string>("base_name") + "_" : ""),
+    _x(declareVector("x")),
+    _y(declareVector("y")),
+    _z(declareVector("z")),
+    _position(declareVector("id")),
+    // get the property name instead of materialname
+    _avg_crack_tip_scalar(declareVector("crack_tip_" + _base_name + _property_name))
+{
+  if (_mesh.dimension() == 3 && !isParamSetByUser("box_width"))
+    paramError("box_width", "Must define box_width in 3D problems.");
+  // add user object dependencies by name (the UOs do not need to exist yet for this)
+  _depend_uo.insert(getParam<UserObjectName>("crack_front_definition"));
+}
+
+void
+CrackFrontNonlocalMaterialBase::initialSetup()
+{
+  // gather coupled user objects late to ensure they are constructed. Do not add them as
+  // dependencies (that's already done in the constructor).
```

Comment:
I would suggest deleting these comments. This is something that I think is generally understood, and applies to virtually every MOOSE object with these types of dependencies.